### ISSUE TITLE
[blutopia] Found no results while trying to browse this tracker (Test)

### ISSUE THREAD

**Issue by DeltaHit** [Jackett/Jackett#1819]
Jackett version:  0.8.175.0
Mono version (if not using Windows):  

-------------------------------
I think Blutopia updated their website and since then, the tracker doesn't seem to work, and it seems to have disappeared from the list of available trackers.

**Comment by kaso17** [comment]
@adamwinn can you help?

**Comment by adamwinn** [comment]
They're not doing a normal form submit to perform a search like all the other trackers. They are now using an 'onkeypressdown' ajax call and it's returning json.  I don't know how to incorporate that into Jackett's existing framework.

The call is something like this:
```
https://blutopia.xyz/filter?_token=<token>&search=spiderman&imdb=&tvdb=&tmdb=&sorting=created_at&direction=desc&qty=25
```

What's returned:
```
{
	"result": ["<tr><td><\/td>\n            <td><a href='https:\/\/blutopia.xyz\/categories\/movies.1'><center><i class='fa fa-film torrent-icon' data-toggle='tooltip' title='' data-original-title='Movie Torrent'><\/i><\/center><\/a><\/td>\n            <td><a class='view-torrent' data-id='3635' data-slug='spider-man-3-2007-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n' href='https:\/\/blutopia.xyz\/torrents\/spider-man-3-2007-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n.3635' data-toggle='tooltip' title='' data-original-title='Spider-Man 3 2007 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1 AVC-LEGi0N'>Spider-Man 3 2007 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1 AVC-LEGi0N<\/a>\n                <a href='https:\/\/blutopia.xyz\/download_check\/spider-man-3-2007-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n.3635'><button class='btn btn-primary btn-circle' type='button' data-toggle='tooltip' title='' data-original-title='DOWNLOAD!'><i class='fa fa-download'><\/i><\/button><\/a>\n\n                <ul><li>\n                <span class='label label-success'>Remux<\/span>\n                <strong>\n                <span class='badge-extra text-bold'>\n                <i class='fa fa-upload'><\/i> By <a href='https:\/\/blutopia.xyz\/members\/swooper.4330'>swooper<\/a>\n                <\/span>\n\n                <a rel='nofollow' href='https:\/\/anon.to?http:\/\/www.imdb.com\/title\/tt0413300'><span class='badge-extra text-black'><i class='fa fa-imdb' data-toggle='tooltip' title='' data-original-title='View IMDB'><\/i><\/span><\/a>\n\n                <span class='badge-extra text-bold text-pink'><i class='fa fa-heart' data-toggle='tooltip' title='' data-original-title='Thanks Given'><\/i>0<\/span>\n\n                <span class='badge-extra text-bold'><i class='fa fa-globe text-blue' data-toggle='tooltip' title='' data-original-title='Global FreeLeech'><\/i> Global FreeLeech<\/span><span class='badge-extra text-bold'><i class='fa fa-globe text-green' data-toggle='tooltip' title='' data-original-title='Double Upload'><\/i> Global Double Upload<\/span><span class='badge-extra text-bold'><i class='fa fa-tachometer text-red' data-toggle='tooltip' title='' data-original-title='High Speeds!'><\/i> High Speeds!<\/span>\n                <\/strong>\n                <\/li><\/ul>\n            <\/td>\n\n            <td><time datetime='2017-09-14 17:09:12'>4 hours ago<\/time><\/td>\n            <td><span class='badge-extra text-blue text-bold'>36.85 GB<\/span><\/td>\n            <td><span class='badge-extra text-orange text-bold'>1 times<\/span><\/td>\n            <td><span class='badge-extra text-green text-bold'>2<\/span><\/td>\n            <td><span class='badge-extra text-red text-bold'>1<\/span><\/td>\n            <\/tr>\n            ", "<tr><td><\/td>\n            <td><a href='https:\/\/blutopia.xyz\/categories\/movies.1'><center><i class='fa fa-film torrent-icon' data-toggle='tooltip' title='' data-original-title='Movie Torrent'><\/i><\/center><\/a><\/td>\n            <td><a class='view-torrent' data-id='3634' data-slug='spider-man-2-spider-man-2-2004-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n' href='https:\/\/blutopia.xyz\/torrents\/spider-man-2-spider-man-2-2004-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n.3634' data-toggle='tooltip' title='' data-original-title='Spider-Man 2 Spider-Man 2 2004 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1 AVC-LEGi0N'>Spider-Man 2 Spider-Man 2 2004 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1 AVC-LEGi0N<\/a>\n                <a href='https:\/\/blutopia.xyz\/download_check\/spider-man-2-spider-man-2-2004-4k-remastered-remux-1080p-bluray-dts-hd-ma-51-avc-legi0n.3634'><button class='btn btn-primary btn-circle' type='button' data-toggle='tooltip' title='' data-original-title='DOWNLOAD!'><i class='fa fa-download'><\/i><\/button><\/a>\n\n                <ul><li>\n                <span class='label label-success'>Remux<\/span>\n                <strong>\n                <span class='badge-extra text-bold'>\n                <i class='fa fa-upload'><\/i> By <a href='https:\/\/blutopia.xyz\/members\/swooper.4330'>swooper<\/a>\n                <\/span>\n\n                <a rel='nofollow' href='https:\/\/anon.to?http:\/\/www.imdb.com\/title\/tt0316654'><span class='badge-extra text-black'><i class='fa fa-imdb' data-toggle='tooltip' title='' data-original-title='View IMDB'><\/i><\/span><\/a>\n\n                <span class='badge-extra text-bold text-pink'><i class='fa fa-heart' data-toggle='tooltip' title='' data-original-title='Thanks Given'><\/i>0<\/span>\n\n                <span class='badge-extra text-bold'><i class='fa fa-globe text-blue' data-toggle='tooltip' title='' data-original-title='Global FreeLeech'><\/i> Global FreeLeech<\/span><span class='badge-extra text-bold'><i class='fa fa-globe text-green' data-toggle='tooltip' title='' data-original-title='Double Upload'><\/i> Global Double Upload<\/span><span class='badge-extra text-bold'><i class='fa fa-tachometer text-red' data-toggle='tooltip' title='' data-original-title='High Speeds!'><\/i> High Speeds!<\/span>\n                <\/strong>\n                <\/li><\/ul>\n            <\/td>\n\n            <td><time datetime='2017-09-14 17:09:57'>4 hours ago<\/time><\/td>\n            <td><span class='badge-extra text-blue text-bold'>33.23 GB<\/span><\/td>\n            <td><span class='badge-extra text-orange text-bold'>1 times<\/span><\/td>\n            <td><span class='badge-extra text-green text-bold'>2<\/span><\/td>\n            <td><span class='badge-extra text-red text-bold'>1<\/span><\/td>\n            <\/tr>\n            ", "<tr><td><\/td>\n            <td><a href='https:\/\/blutopia.xyz\/categories\/movies.1'><center><i class='fa fa-film torrent-icon' data-toggle='tooltip' title='' data-original-title='Movie Torrent'><\/i><\/center><\/a><\/td>\n            <td><a class='view-torrent' data-id='3633' data-slug='spider-man-2002-4k-remastered-remux-1080p-bluray-dts-hd-ma-51avc-legi0n' href='https:\/\/blutopia.xyz\/torrents\/spider-man-2002-4k-remastered-remux-1080p-bluray-dts-hd-ma-51avc-legi0n.3633' data-toggle='tooltip' title='' data-original-title='Spider-Man 2002 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1.AVC-LEGi0N'>Spider-Man 2002 4K-REMASTERED REMUX 1080p BluRay DTS-HD MA 5.1.AVC-LEGi0N<\/a>\n                <a href='https:\/\/blutopia.xyz\/download_check\/spider-man-2002-4k-remastered-remux-1080p-bluray-dts-hd-ma-51avc-legi0n.3633'><button class='btn btn-primary btn-circle' type='button' data-toggle='tooltip' title='' data-original-title='DOWNLOAD!'><i class='fa fa-download'><\/i><\/button><\/a>\n\n                <ul><li>\n                <span class='label label-success'>Remux<\/span>\n                <strong>\n                <span class='badge-extra text-bold'>\n                <i class='fa fa-upload'><\/i> By <a href='https:\/\/blutopia.xyz\/members\/swooper.4330'>swooper<\/a>\n                <\/span>\n\n                <a rel='nofollow' href='https:\/\/anon.to?http:\/\/www.imdb.com\/title\/tt0145487'><span class='badge-extra text-black'><i class='fa fa-imdb' data-toggle='tooltip' title='' data-original-title='View IMDB'><\/i><\/span><\/a>\n\n                <span class='badge-extra text-bold text-pink'><i class='fa fa-heart' data-toggle='tooltip' title='' data-original-title='Thanks Given'><\/i>1<\/span>\n\n                <span class='badge-extra text-bold'><i class='fa fa-globe text-blue' data-toggle='tooltip' title='' data-original-title='Global FreeLeech'><\/i> Global FreeLeech<\/span><span class='badge-extra text-bold'><i class='fa fa-globe text-green' data-toggle='tooltip' title='' data-original-title='Double Upload'><\/i> Global Double Upload<\/span><span class='badge-extra text-bold'><i class='fa fa-tachometer text-red' data-toggle='tooltip' title='' data-original-title='High Speeds!'><\/i> High Speeds!<\/span>\n                <\/strong>\n                <\/li><\/ul>\n            <\/td>\n\n            <td><time datetime='2017-09-14 17:09:17'>5 hours ago<\/time><\/td>\n            <td><span class='badge-extra text-blue text-bold'>31.54 GB<\/span><\/td>\n            <td><span class='badge-extra text-orange text-bold'>1 times<\/span><\/td>\n            <td><span class='badge-extra text-green text-bold'>2<\/span><\/td>\n            <td><span class='badge-extra text-red text-bold'>2<\/span><\/td>\n            <\/tr>\n            "],
	"rows": 3,
	"qty": "25",
	"active": 1
}
```

The good news is they got rid of forcing the user to choose BD50 and those other options that I had to hardcode in earlier

**Comment by kaso17** [comment]
That's HTML encapsulated in JSON.
If you change the row selector to just "tr", does it result in any found rows (the anglesharp parser might simply ignore the surrounding JSON garbage)?
```
    rows:
      selector: tr
```

**Comment by adamwinn** [comment]
Ya, I know it is :)

What am I supposed to put for this section?  None of it even applies anymore.

```
  search:
    path: /torrents/search
    inputs:
      $raw: "{{range .Categories}}filter_cat[{{.}}]=1&{{end}}"
      name: "{{ .Query.Keywords }}"
      category_id: 1
      type: BD50
      order: created_at:desc
```

**Comment by kaso17** [comment]
you have to adjust it do the new search URL:
https://blutopia.xyz/filter?_token=<token>&=spiderman&imdb=&tvdb=&tmdb=&sorting=created_at&direction=desc&qty=25
```
  search:
    path: /filter
    inputs:
      _token: <token>
      search: "{{ .Keywords }}"
      imdb: "{{ .Query.IMDBID }}"
      sorting=created_at
      direction=desc
      qty=100
```

**Comment by adamwinn** [comment]
@kaso17 How do I get the token?

**Comment by kaso17** [comment]
What kind of token is it?
An static one used for authentication?
A random anti cross site scripting token?

Did you check if it's really needed?

**Comment by adamwinn** [comment]
I don't know the answer to those questions.  I never use the site.

**Comment by kaso17** [comment]
Someone has an invite so i can have a look?

**Comment by adamwinn** [comment]
@HDVinnie might have one

**Comment by HDVinnie** [comment]
can email me at blutopia@stealth.tg if need invite. But to answer question @kaso17 @adamwinn its a laravel csrf token.

**Comment by kaso17** [comment]
email sent.

If it's a changing CSRF token we definitely will need a C# implementation.

**Comment by Killswitch11** [comment]
Currently under open signup https://blutopia.xyz/register

**Comment by kaso17** [comment]
I got the invite , thank you will work on this soon.

**Comment by HDVinnie** [comment]
can't you just do this?

```
---
  site: blutopia
  name: Blutopia
  description: "HD Movie/TV/FANRES Tracker"
  language: en-us
  type: private
  encoding: UTF-8
  links:
    - https://blutopia.xyz/

  caps:
    categorymappings:
      - {id: 1, cat: Movies, desc: "Movies"}
      - {id: 2, cat: TV, desc: "TV"}
      - {id: 3, cat: Movies, desc: "FANRES"}

    modes:
      search: [q]

  login:
    path: /login
    method: form
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
    error:
      - selector: table.main:contains("Login Failed!")
    test:
      path: /torrents

  download:
    selector: a[href^="/download.php/"]
        
  search:
    path: /filter
    inputs:
      _token=qON8IXNrnS6lTyUTDZUeKmY5jkQG05OxXoUzaEp9
      search: "{{ .Keywords }}"
      imdb: "{{ .Query.IMDBID }}"
      sorting=created_at
      direction=desc
      qty=100
    rows:
      selector: table.table > tbody > tr
    fields:
      # category:
      #   selector: a[href^="categories"]
      #   attribute: href
      title:
        remove: ul
        selector: td:nth-child(2)
      download:
        selector: a.view-torrent
        attribute: href
        filters:
          - name: replace
            args: ["/torrents/", "/download/"]
      details:
        selector: a.view-torrent
        attribute: href
      size:
        selector: td:nth-child(4)
      seeders:
        selector: td:nth-child(6)
      leechers:
        selector: td:nth-child(7)
      grabs:
        selector: td:nth-child(5)
        filters:
          - name: regexp
            args: (\d+)
      date:
        selector: td:nth-child(3)
        filters:
          - name: append
            args: " ago"
      downloadvolumefactor:
        case:
          #"i[data-original-title=\"100% Free\"]": "0"
          "*": "1"
      uploadvolumefactor:
        case:
          #"i[data-original-title=\"Double upload\"]": "2"
          "*": "1"
```

**Comment by seanvree** [comment]
I'm getting the same error
Windows 10E x64 1703
Jackett Version 0.8.225.0

```
2017-09-26 00:39:51.8300 Debug IWebClient.GetString(Url:https://blutopia.xyz/torrents/search?name=&category_id=1&type=BD50&order=created_at%3adesc) 
2017-09-26 00:39:52.3148 Debug IWebClient: Returning NotFound => <!DOCTYPE html>
<html class="no-js" lang="en">

<head>
  <meta charset="utf-8">
  <title>Error 404: Page Not Found - Blutopia</title>

  <!-- Meta -->
        <meta http-equiv="x-ua-compatible" content="ie=edge">
      <meta name="viewport" content="width=device-width, initial-scale=1">
      <meta name="description" content="">
      <meta property="og:title" content="Blutopia">
      <meta property="og:type" content="website">
      <meta property="og:image" content="https://blutopia.xyz/img/rlm.png">
      <meta property="og:url" content="https://blutopia.xyz">
      <meta name="csrf-token" content="">
    <!-- /Meta -->

  <!--icons -->
  <link rel="shortcut icon" href="https://blutopia.xyz/favicon.ico" type="image/x-icon">
  <link rel="icon" href="https://blutopia.xyz/favicon.ico" type="image/x-icon">
  <!--icons -->

  <!--css -->
  <link rel="stylesheet" href="https://blutopia.xyz/css/main/custom.css">
  <link rel="stylesheet" href="https://blutopia.xyz/css/main/advbuttons.css">
  <link rel="stylesheet" href="https://blutopia.xyz/css/vendor/vendor.min.css" />
  <!--css -->
  
</head>

<body>
  <section class="container content" id="content-area" style="min-height: 344px;">
    <h2></h2>
    <div class="jumbotron shadowed">
      <div class="container">
        <h1 class="mt-5 text-center">
          <i class="fa fa-question-circle text-warning"></i> Error 404: Page Not Found
        </h1>
        <div class="separator"></div>
        <p class="text-center">The Requested Page Cannot Be Found! <br>Not Sure What Your Looking For But Check The Address And Try Again!</p>
        <p class="text-center">
          <a href="https://blutopia.xyz" role="button" class="btn btn-labeled btn-primary"><span class="btn-label"><i class="fa fa-home"></i></span>Go Home</a>
        </p>
      </div>
    </div>
  </section>
</body>

</html>
```

**Comment by HDVinnie** [comment]
@seanvree thats because Blutopia no longer uses that search system
https://blutopia.xyz/torrents/search?name=&category_id=1&type=BD50&order=created_at%3adesc

read above its now
https://blutopia.xyz/filter?_token= qON8IXNrnS6lTyUTDZUeKmY5jkQG05OxXoUzaEp9&search=&imdb=&tvdb=&tmdb=&sorting=created_at&direction=desc&qty=25

**Comment by kaso17** [comment]
fixed via 48118d569169e018c0cd6c8013383a6b9744bad7
